lib/tx.py

```diff
@@ -217,3 +217,11 @@ def read_tx(self):
 
         return TxSegWit(version, marker, flag, inputs,
                         outputs, witness, locktime), double_sha256(orig_ser)
+
+
+class DeserializerFairCoin(Deserializer):
+    def read_block(self):
+        '''Returns a list of (deserialized_tx, tx_hash) pairs.'''
+        read_tx = self.read_tx
+        txs = [read_tx() for n in range(self._read_varint())]
+        return txs
```

Comment:
Is this deserializer necessary?

The code seems the same as the `Deserializer::read_block` with the `assert self.cursor == len(self.binary)` check. Did you get a crash on that assert?